### DISCUSSION TITLE
gslc_ElemSetTxtStr not updating text or crash

### DISCUSSION THREAD

**Discussion by claudchtiscali** [ImpulseAdventure/GUIslice#481] [Q&A] [answered]
I'm trying to update the text of a dynamic text field created in GUISlice Builder. 
If I set the MAX length for the text to update to 0 the code displays the GUI but does not update the text element on the GUI (of course) but setting the length to any number the CPU crashes with the following message :
Guru Meditation Error: Core  1 panic'ed (LoadStoreError). Exception was unhandled.

Core  1 register dump:
PC      : 0x400d1859  PS      : 0x00060c30  A0      : 0x800d196e  A1      : 0x3ffb27a0  
A2      : 0x3f40017e  A3      : 0x3f400267  A4      : 0x00000005  A5      : 0x3ffc1b30  
A6      : 0x00000000  A7      : 0x3f40017e  A8      : 0x00000031  A9      : 0x3ffb2780  
A10     : 0x00000003  A11     : 0x00000001  A12     : 0x0000000a  A13     : 0x00000001  
A14     : 0x00ff0000  A15     : 0xff000000  SAR     : 0x00000004  EXCCAUSE: 0x00000003  
EXCVADDR: 0x3f40017e  LBEG    : 0x40086225  LEND    : 0x40086235  LCOUNT  : 0xfffffffa  

Backtrace:0x400d1856:0x3ffb27a00x400d196b:0x3ffb27d0 0x400d1a2c:0x3ffb2800 0x400dadb9:0x3ffb2820 

I'm using a ESP32 with ILI9341
I've ran a little debugging and I found that the problem is in the strncpy () in the gslc_StrCopy() called by gslc_ElemSetTxtStr
[ILI9341_T.zip](https://github.com/ImpulseAdventure/GUIslice/files/9944404/ILI9341_T.zip)

Could someone help me ... ? 
Thanks in advance.

**Comment by claudchtiscali** [comment, marked as answer]
It's definitely an issue of the library for ESP32, if I compile for Arduino Mega the same code run perfectly. 
I'll move this topic to "issues"

**Comment by Pconti31** [comment]
@claudchtiscali A couple of things, first I suggest you post in issues when you need help.    I myself don't check here very often.
Next time it would help me if you include your project file *.prj when posting zip files so I can check your definitions.
This also helps when I don't have similar hardware and need to make changes for what I do have.

In this case in looking over your header I noticed no  modifiable storage was allocated to your text field.  
```
  pElemRef = gslc_ElemCreateTxt(&m_gui,E_ELEM_TEXT7,E_PG_MAIN,(gslc_tsRect){190,70,43,22},
    (char*)"Spento  ",10,E_DOSIS_BOOK8);
```
Note that your gslc_ElemSetTxtStr(&m_gui, pElemCnt, a); statement is attempting to write over constant flash storage (non-writable) memory containing "Spento" memory .  It's somewhat scary that Arduino mega allows that...
Your field should have look like:
```
  static char m_sDisplayText7[11] = "";
  pElemRef = gslc_ElemCreateTxt(&m_gui,E_ELEM_TEXT7,E_PG_MAIN,(gslc_tsRect){190,70,43,22}
    ,m_sDisplayText7,11,E_DOSIS_BOOK8);
```
and when set this way your program works fine with esp32.
Without your project file I can't fix this for you but your field inside the builder should be defined like this more or less:

![text-input](https://user-images.githubusercontent.com/44075106/200925958-b85d6bf5-bb5a-48da-a358-c6a1d2bc9ff6.png)

Given the field ElementRef=m_pElementCnt you don't need to do the lookup call
```
gslc_tsElemRef*  pElemCnt        = gslc_PageFindElemById(&m_gui,E_PG_MAIN,E_ELEM_TEXT7);
```
and can instead directly reference:
```
gslc_ElemSetTxtStr(&m_gui,m_pElemCnt, a);
```

**Comment by Pconti31** [comment]
I feel that for other people I should point out that this issue is common enough that I included a section for it in my faq.pdf supplied with the Builder releases titled: **Can you change a Text String at runtime?** that explains how to set the text field inside your project.
You should always check there first before posting questions.
Paul--
